src/main/java/jimmy/mcgymmy/logic/predicate/DatePredicate.java

```diff
@@ -0,0 +1,44 @@
+package jimmy.mcgymmy.logic.predicate;
+
+import java.util.function.Predicate;
+import java.util.logging.Logger;
+
+import jimmy.mcgymmy.commons.core.LogsCenter;
+import jimmy.mcgymmy.logic.parser.ParserUtil;
+import jimmy.mcgymmy.logic.parser.exceptions.ParseException;
+import jimmy.mcgymmy.model.date.Date;
+import jimmy.mcgymmy.model.food.Food;
+
+/**
+ * Tests that a {@code Food}'s {@code Date} matches the {@code String} date given.
+ */
+public class DatePredicate implements Predicate<Food> {
```

Comment:
A suggestion I have is to collect all these `Predicate` classes into 1 file and turn each individual `Predicate` class into a function that returns an anonymous class, i.e.

```java
public class Predicates {
  public Predicate<Food> makeDatePredicate(String date) throws ParseException {
    return new Predicate<Food>() {
      // all the code below
      }
    }
  // ... more predicates
}
```

Will reduce a lot of the java boilerplate in the codebase.

Reply:
Good idea, might do this.